libraries/Microsoft.Bot.Builder.Core/BotContext.cs

```diff
@@ -39,78 +39,97 @@ public BotContext(BotAdapter bot, ConversationReference conversationReference)
             _adapter = bot ?? throw new ArgumentNullException(nameof(bot));
             _conversationReference = conversationReference ?? throw new ArgumentNullException(nameof(conversationReference));
         }
-
-        //public async Task SendActivity(IBotContext context, IList<IActivity> activities)
-        //{
-        //    await _bot.SendActivity(context, activities).ConfigureAwait(false);
-        //}
-
-        public IActivity Request => _request;
-
         public BotAdapter Adapter => _adapter;
 
-        public IList<IActivity> Responses { get => _responses; set => this._responses = value; }
+        public Activity Request => _request;
 
-        public IStorage Storage { get; set; }
+        public IList<Activity> Responses { get => _responses; set => this._responses = value; }
 
-        public Intent TopIntent { get; set; }
-
-        public TemplateManager TemplateManager { get; set; }
+        public ConversationReference ConversationReference { get => _conversationReference; }
 
-        public bool IfIntent(string intentName)
+        public IBotContext Reply(string text, string speak = null)
         {
-            if (string.IsNullOrWhiteSpace(intentName))
-                throw new ArgumentNullException(nameof(intentName));
-
-            if (this.TopIntent != null)
+            var reply = this.ConversationReference.GetPostToUserMessage();
+            reply.Text = text;
+            if (!string.IsNullOrWhiteSpace(speak))
             {
-                if (TopIntent.Name == intentName)
-                {
-                    return true;
-                }
+                // Developer included SSML to attach to the message.
+                reply.Speak = speak;
             }
+            this.Responses.Add(reply);
+            return this;
+        }
 
-            return false;
+        public IBotContext Reply(IActivity activity)
+        {
+            BotAssert.ActivityNotNull(activity);
+            this.Responses.Add((Activity)activity);
+            return this;
         }
-        public bool IfIntent(Regex expression)
+
+        public void Set(string serviceId, object service)
         {
-            if (expression == null)
-                throw new ArgumentNullException(nameof(expression));
+            if (String.IsNullOrWhiteSpace(serviceId))
+                throw new ArgumentNullException(nameof(serviceId));
+            lock (_services)
+            {
+                this._services[serviceId] = service;
+            }
+        }
 
-            if (this.TopIntent != null)
+        public object Get(string serviceId)
```

Comment:
Why custom semantics? Seems like standard dictionary semantics would be the way to go. 

That is, rather than TryGetValue, just: 
    return this._services[serviceId]

That would give developers strandard dictioanry semantics, which is.. normal.